packages/mdc-textfield/README.md

```diff
@@ -153,9 +153,9 @@ by HTML5's form validation API.
 </div>
 ```
 
-An input's validity is checked via `checkValidity()` on blur, and the styles are updated
-accordingly. When using the `required` attribute, an asterisk will be automatically appended to the
-label text, as per the spec.
+By default an input's validity is checked via `checkValidity()` on blur, and the styles are updated
+accordingly. The validity can also be set via the MDCTextfield.valid variable. When using the `required`
```

Comment:
Try this:

Set the MDCTextfield.valid variable to set the input's validity explicitly. MDC Textfield automatically appends an asterisk to the label text if the `required` attribute is set.